File: 3016-minimum-number-of-pushes-to-type-word-ii/3016-minimum-number-of-pushes-to-type-word-ii.cpp
using namespace std;
#include <unordered_map>
#include <string>
#include <algorithm>
#include <vector>
class Solution {
public:
    int minimumPushes(string word) {
        vector<int> cnt(26);
        for (const auto& x : word) {
            ++cnt[x - 'a'];
        }
        sort(rbegin(cnt), rend(cnt));
        int result = 0;
        for (int i = 0; i < size(cnt); ++i) {
            result += cnt[i] * (i / (9 - 2 + 1) + 1);
        }
        return result;
    }
};

File: 3016-minimum-number-of-pushes-to-type-word-ii/3016-minimum-number-of-pushes-to-type-word-ii.py
from collections import Counter

class Solution:
    def minimumPushes(self, word: str) -> int:
        freq = Counter(word)
        
        sorted_freq = sorted(freq.values(), reverse=True)
        n = len(sorted_freq)
        
        
        num_keys = 8
        total_presses = 0
        press_cost = 1
        
        for i in range(n):
            key_press_number = i // num_keys + 1
            total_presses += sorted_freq[i] * key_press_number
        
        return total_presses